engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -1237,29 +1397,63 @@ public Class<?> visit(UnaryExpr n, VisitArgs printer) {
             throw new RuntimeException("Unary operation (" + n.getOperator().name() + ") not supported");
         }
 
-        printer.append(opName).append('(');
-        Class<?> type = n.getExpression().accept(this, printer);
-        printer.append(')');
+        final Class<?> ret = getTypeWithCaching(n.getExpression());
 
-        return getMethodReturnType(null, opName, new Class[] {type}, getParameterizedTypes(n));
+        final boolean isNonequalOpOverload =
+                n.getComment().orElse(null) == QueryLanguageParserComment.BINARY_OP_NEQ_CONVERSION_FLAG;
+        if (isNonequalOpOverload) {
+            Assert.equals(ret, "ret", boolean.class, "boolean.class");
+        }
+
+        if (boolean.class.equals(ret)) {
+            // primitive booleans don't need a 'not()' method; they can't be null anyway
+            printer.append('!');
+
+            final Class<?> retForCheck = n.getExpression().accept(this, printer);
+            Assert.equals(retForCheck, "retForCheck", ret, "ret");
+
+            if (isNonequalOpOverload && printer.hasStringBuilder()) {
+                // sanity checks -- the inner expression *must* be a BinaryExpr (for ==), and it must be replaced in
+                // this
+                // UnaryExpr with a MethodCallExpr (for "eq()" or possibly "isNull()").
+                Assert.instanceOf(n.getExpression(), "n.getExpression()", MethodCallExpr.class);
+                final MethodCallExpr methodCall = (MethodCallExpr) n.getExpression();
+                final String methodName = methodCall.getNameAsString();
+                if (!"eq".equals(methodName) && !"isNull".equals(methodName)) {
+                    throw new IllegalStateException(
+                            "Unexpected node with BINARY_OP_NEQ_CONVERSION_FLAG: " + methodCall);
+                }
+            }
+
+            return ret;
+        } else {
+            Node origParent = n.getParentNode().orElseThrow();
+            MethodCallExpr unaryOpOverloadMethod =
+                    new MethodCallExpr(null, opName, NodeList.nodeList(n.getExpression()));
```

Comment:
There is a MethodCallExpr constructor overload that takes care of the null param for you.
```suggestion
                    new MethodCallExpr(opName, NodeList.nodeList(n.getExpression()));
```